### DISCUSSION TITLE
When is the render data refreshed?

### DISCUSSION THREAD

**Discussion by matiascodesal** [GPUOpen-LibrariesAndSDKs/BlenderUSDHydraAddon#23] [Q&A]
It's unclear to me when the plugin re-reads changes from the input USD files and blender data?  I found myself needing to rebuild the node tree in order to see my changes in the viewport.

**Comment by bnagirniak** [comment]
Yes, in some cases you have to rebuild node tree to see changes in viewport, this part will be fixed/improved.

**Comment by bsavery** [comment]
Can you convert this into an issue with some examples?  @matiascodesal

**Comment by matiascodesal** [comment]
Happy New Year!  Here's the logged issue: https://github.com/GPUOpen-LibrariesAndSDKs/BlenderUSDHydraAddon/issues/30

Another issue that threw me off early on is that if your USD stage doesn't have a defaultPrim defined, the "Read USD File" node will error and not read the USD file.  This is unfortunate because Blender's USD export writes a file with multiple prims at the root level and no defaultPrim.  This most basic workflow fails:
1. Export a Blender scene to USD
2. Load it with BlenderUSDHydra

This is probably more of an issue with the exporter though?
